ocaml/typing/mode.ml

```diff
@@ -85,8 +87,15 @@ module Lattices = struct
 
     let legacy = Global
 
-    let le a b =
-      match a, b with Global, _ | _, Local -> true | Local, Global -> false
+    let less_or_equal a b : Misc.Le_result.t =
+      match a, b with
+      | Global, Global -> Equal
+      | Global, _ -> Less
+      | Local, Local -> Equal
+      (* | _, Local -> Less *)
```

Comment:
When/if we add a third mode, it will no longer be redundant. I'm happy to explain why I've kept it, but keeping was intentional. I don't feel strongly here.